src/index.ts

```diff
@@ -3,64 +3,98 @@ import path from 'path'
         localNetworksConfig.defaultConfig,
-        localNetworkConfig
+        localNetworkConfig as object,
+      ]) as NetworkConfig)
+    }
+  })
+
+  Object.entries(homeLocalNetworksConfig.networks).forEach(([networkName, localNetworkConfig]) => {
+    if (!hardhatConfig.networks[networkName]) {
+      hardhatConfig.networks[networkName] = (deepmerge.all([
+        hardhatConfig.networks[networkName] || {},
+        homeLocalNetworksConfig.defaultConfig,
+        localNetworksConfig.defaultConfig,
+        localNetworkConfig as object,
       ]) as NetworkConfig)
     }
   })
 });
 
-export function readLocalNetworksConfig(userConfig: HardhatUserConfig): LocalNetworksConfig {
-  const localNetworksConfigPath = parseLocalNetworksConfigPath(userConfig)
-  const localNetworksConfig = localNetworksConfigPath ? require(localNetworksConfigPath) : {}
+export function readHomeNetworksConfig(): LocalNetworksConfigInternal {
+  const configPath = getDefaultHomeLocalNetworksConfigPath()
+  const networksConfig = fs.existsSync(configPath) ? require(configPath) : {}
 
-  if (!localNetworksConfig.networks) localNetworksConfig.networks = []
-  if (!localNetworksConfig.defaultConfig) localNetworksConfig.defaultConfig = {}
+  if (!networksConfig.networks) networksConfig.networks = []
+  if (!networksConfig.defaultConfig) networksConfig.defaultConfig = {}
 
-  return localNetworksConfig
+  return networksConfig
 }
 
-export function parseLocalNetworksConfigPath(userConfig: HardhatUserConfig): string | undefined {
-  const localNetworksConfigPath = userConfig.localNetworksConfig
-  if (typeof localNetworksConfigPath === 'string' && fs.existsSync(localNetworksConfigPath)) {
-    return localNetworksConfigPath
+export function readLocalNetworksConfig(
+  hardhatConfig: HardhatConfig,
+  userConfig: HardhatUserConfig
+): LocalNetworksConfigInternal {
+  const localNetworksConfigPath = parseLocalNetworksConfigPath(userConfig, hardhatConfig.paths.root)
+
+  if (localNetworksConfigPath && !fs.existsSync(localNetworksConfigPath)) {
+    throw new HardhatPluginError(
+      `hardhat-local-networks-config-plugin`,
+      `configuration file not found under "localNetworksConfig" path: ${userConfig.localNetworksConfig}; ` +
+        `resolved path: ${localNetworksConfigPath}`
+    )
   }
 
-  const defaultLocalNetworksConfigPath = getDefaultLocalNetworksConfigPath()
-  return fs.existsSync(defaultLocalNetworksConfigPath) ? defaultLocalNetworksConfigPath : undefined
+  const localNetworksConfig = localNetworksConfigPath ? require(localNetworksConfigPath) : {}
+
+  if (!localNetworksConfig.networks) localNetworksConfig.networks = []
+  if (!localNetworksConfig.defaultConfig) localNetworksConfig.defaultConfig = {}
+
+  return localNetworksConfig
 }
 
-export function getDefaultLocalNetworksConfigPath() {
-  return path.join(getLocalConfigDir(), HARDHAT_NETWORK_CONFIG_FILE)
+export function getDefaultHomeLocalNetworksConfigPath() {
```

Comment:
What do you think about?

```suggestion
export function getHomeNetworksConfigPath() {
```